server/src/main/java/com/google/musicanalysis/site/MusixParamBuilder.java

```diff
@@ -0,0 +1,87 @@
+package com.google.musicanalysis.site;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+
+/*
+    Builds parameter strings based off what the user wants.
+    This is structured the way it is becuase I will be adding more
+    to each function. I will clean it up later as well.
+*/
+public class MusixParamBuilder {
+
+  private String intendedUse;
+
+  // All possible parameters I can receive
+  private String trackId;
+  private String trackName;
+  private String albumId;
+  private String artistName;
+
+  /**
+   * Constructor.
+   *
+   * @param intendedUse determines what action we will send to the API
+   * @param params array of parameters which get assigned to different values depending on
+   *     intendedUse
+   */
+  public MusixParamBuilder(String intendedUse, String... params) {
+    this.intendedUse = intendedUse;
+
+    switch (intendedUse) {
+      case "track.search":
+        this.trackName = params[0];
+        this.artistName = params[1];
+        break;
+      case "track.lyrics.get":
+        this.trackId = params[0];
+        break;
+      default:
+        throw new IllegalArgumentException("arguments required");
+    }
+  }
+
+  /** Filters out the correct parameter string and returns it */
+  public String filterParamString() {
+    switch (this.intendedUse) {
+      case "track.search":
+        return buildTrackString(this.trackName, this.artistName);
+      case "track.lyrics.get":
+        return buildLyricsString(this.trackId);
+    }
+    return null;
+  }
+
+  /**
+   * Builds string to serve to request (track search)
+   *
+   * @param trackName name of the desired track
+   * @param artistName name of the desired artist
+   */
+  private String buildTrackString(String trackName, String artistName) {
+    // ready string for sending
+    try {
+      trackName = URLEncoder.encode(trackName, "UTF-8");
+      artistName = URLEncoder.encode(artistName, "UTF-8");
+    } catch (UnsupportedEncodingException err) {
+      throw new AssertionError("Invalid URL");
+    }
```

Comment:
I discovered after making the earlier suggestion that you can use `StandardCharsets.UTF_8` instead of `"UTF-8"`, which will save you from needing the try-catch block. It's not a must-change, but if you want to simplify this.

You can import it like this:
```java
import java.nio.charset.StandardCharsets;
```

```suggestion
    trackName = URLEncoder.encode(trackName, StandardCharsets.UTF_8);
    artistName = URLEncoder.encode(artistName, StandardCharsets.UTF_8);
```

Reply:
Sounds good to me